### PR TITLE
feat: use of agent_name in RESTclient (agent_exists,get_agent)

### PR DESCRIPTION
**Pull request by 2mmanu**
**Please describe the purpose of this pull request.**

The pull request introduces a feature in the RESTClient for the methods 'agent_exists' and 'get_agent'. In the previous version, the parameter 'agent_name' was not used properly.

**How to test**
**Have you tested this PR?**
I tested it myself and integrated two assertions in the client test.

**Related issues or PRs**
na

**Is your PR over 500 lines of code?**
nope

**Additional context**
na

### CODE REVIEW

File: memgpt/client/client.py [hunk 1]
```diff
@@ -236,7 +236,22 @@ def list_agents(self):
         response = requests.get(f"{self.base_url}/api/agents", headers=self.headers)
         return ListAgentsResponse(**response.json())
 
+    def get_agent_id_from_name(self, agent_name: str) -> uuid:
+        agents = self.list_agents()
+        if agents.num_agents > 0:
```
Comment (AlekseyLobanov) [inline memgpt/client/client.py:241]: Changing order of expressions may improve readability.
```python
if not agents.num_agents:
  raise ValueError("no agents available")
agent_name = [agent["id"] for agent in agents.agents if agent["name"] == agent_name]
if not agent_name:
  raise NameError(f"no {agent_name} agent found")
return agent_name[0]
```
Same code, but success path was simplified to less nesting levels. What do you think?

File: memgpt/client/client.py [hunk 2]
```diff
@@ -236,7 +236,22 @@ def list_agents(self):
...
+            agent_name = [agent["id"] for agent in agents.agents if agent["name"] == agent_name]
```
Comment (AlekseyLobanov) [inline memgpt/client/client.py:242]: Maybe it should be named as `agent_ids`. Because of `agent["id"]`

File: memgpt/client/client.py (hunk 1 above)
Comment (2mmanu) [reply memgpt/client/client.py]: I appreciate your comment. With another fix, the code will be as follows:

``` python
def get_agent_id_from_name(self, agent_name: str) -> uuid:
  agents = self.list_agents()
  if not agents.num_agents:
      raise ValueError("no agents available")
  agent_id = next((agent["id"] for agent in agents.agents if agent["name"] == agent_name), None)
  if not agent_id:
      raise NameError(f"No agent named {agent_name} found")
  return agent_id  
```

File: memgpt/client/client.py (hunk 2 above)
Comment (2mmanu) [reply memgpt/client/client.py]: Thank you, but the correct correction would be agent_id. The function will return the first matching name. I am assuming that the names of the agents are unique.

**Review by 2mmanu — COMMENTED** [review]
improve readability

**Pull request closed without merging**